lfs/http.go

```diff
@@ -139,6 +140,21 @@ func (c *Configuration) HttpClient(host string) *HttpClient {
 	return client
 }
 
+func proxyFromGitConfigOrEnvironment(req *http.Request) (*url.URL, error) {
+	proxyURL, err := http.ProxyFromEnvironment(req)
+
+	if proxyURL != nil {
+		return proxyURL, err
```

Comment:
Looks like we return the environment's proxy before checking if there's one in the user's `.gitconfig`. Since this method is called `proxyFromGitConfigOrEnvironment`, the name makes me think that it'll return the `.gitconfig`'s proxy _over_ the environment's one.

Either is fine, but I think it may be beneficial to update the method name if the implementation prioritizes environment configuration over `.gitconfig.`